packages/flutter_tools/test/integration.shard/flutter_build_with_compilation_error_test.dart

```diff
@@ -0,0 +1,80 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.8
+
+import 'package:file/file.dart';
+import 'package:flutter_tools/src/base/io.dart';
+
+import '../src/common.dart';
+import 'test_utils.dart';
+
+void main() {
+  Directory tempDir;
+  Directory projectRoot;
+  String flutterBin;
+  final List<String> targetPlatforms = <String>[
+    'apk',
+    'web',
+    if (platform.isWindows)
+      'windows',
+    if (platform.isLinux)
+      'linux',
+    if (platform.isMacOS)
+      ...<String>['macos', 'ios'],
+  ];
+
+  setUpAll(() {
+    tempDir = createResolvedTempDirectorySync('build_compilation_error_test.');
+    flutterBin = fileSystem.path.join(
+      getFlutterRoot(),
+      'bin',
+      'flutter',
+    );
+
+    processManager.runSync(<String>[
+      flutterBin,
+      ...getLocalEngineArguments(),
+      'create',
+      'hello',
+    ], workingDirectory: tempDir.path);
+
+    projectRoot = tempDir.childDirectory('hello');
+    writeFile(fileSystem.path.join(projectRoot.path, 'lib', 'main.dart'), '''
+int x = 'String';
+''');
+  });
+
+  tearDownAll(() {
+    tryToDelete(tempDir);
+  });
+
+  for (final String targetPlatform in targetPlatforms) {
+    testWithoutContext('flutter build $targetPlatform shows dart compilation error in non-verbose', () {
+      if (targetPlatform == 'linux') {
+        processManager.runSync(<String>[
+          flutterBin,
+          'config',
+          '--enable-linux-desktop',
+        ], workingDirectory: projectRoot.path);
```

Comment:
Matches dev/devicelab behavior to enable desktop features.
https://github.com/flutter/flutter/blob/b0afe3464e7d280f736917528e1c87046d6959f3/dev/devicelab/lib/framework/framework.dart#L120